src/proof_generation/metamath/ast.py

```diff
@@ -152,6 +179,9 @@ class ConstantStatement(Statement):
     def visit(self, visitor: MetamathVisitor[ResultT]) -> ResultT:
         return visitor.proxy_visit_constant_statement(self)  # type: ignore
 
+    def map_inner(self, f: Callable[[Statement], Statement]) -> ConstantStatement:
+        return self
```

Comment:
Why constant statements can't be mapped over? What if you want to change the names?

Reply:
Right now, we have two separate mechanisms for mapping of metamath syntax. The first is to map over statements, the second is to map over terms. Each of these need a different signature `Term -> Term` or `Statement -> Statement`. Combining these would have needed a more complex signature.
We could decide to go that way in the future.

Note that `ConstantStatment`s contain a list strings, constant names, these are neither `Term`s nor `Statement`s and would complicate the signature further. We don't need this at the moment, because  to declare the constants we simply collect all the constants used and add a single statement containing them all.